### ISSUE TITLE
Background Job exits on every occasion.

### ISSUE THREAD

**Issue by Vad1mo** [SUSE/Portus#1756]
Here are some new Stacktraces that I see in the registry.

`MANIFEST_UNKNOWN` is an error in the registry (corrupt layer), however IMHO the background job should continue and not exit on such error or in general on errors.  I am not sure what is the reason for the  stacktrace at `20:03`.  

The backgound job should be really robus and only exit when clear that all further operation will fail. 

```
2018-04-03T17:41:49.707351526Z [Initialization] Running: 'Registry events', 'Security scanning', 'Registry synchronization'
2018-04-03T17:41:52.699825137Z /app/lib/portus/http_helpers.rb:95:in `handle_error':  (Portus::Errors::NotFoundError)
2018-04-03T17:41:52.699921249Z Code: 404
2018-04-03T17:41:52.699945880Z Reported by Registry:
2018-04-03T17:41:52.699968728Z {"code"=>"MANIFEST_UNKNOWN", "message"=>"manifest unknown", "detail"=>{"Name"=>"devops/jenkins", "Revision"=>"sha256:34941ad31dda776de837f76a1ac5a29afea2255dd4dcdfaee1e2bac2a73d26f4"}}
2018-04-03T17:41:52.699998313Z 
2018-04-03T17:41:52.700019958Z Passed values:
2018-04-03T17:41:52.700041632Z repository: devops/jenkins
2018-04-03T17:41:52.700063467Z tag: 664
2018-04-03T17:41:52.700085081Z 
2018-04-03T17:41:52.700106515Z 	from /app/lib/portus/registry_client.rb:67:in `manifest'
2018-04-03T17:41:52.700128771Z 	from /app/lib/portus/security.rb:46:in `vulnerabilities'
2018-04-03T17:41:52.700150666Z 	from /app/lib/portus/background/security_scanning.rb:47:in `block in execute!'
2018-04-03T17:41:52.700172762Z 	from /usr/local/bundle/gems/activerecord-4.2.10/lib/active_record/relation/batches.rb:51:in `block (2 levels) in find_each'
2018-04-03T17:41:52.700195569Z 	from /usr/local/bundle/gems/activerecord-4.2.10/lib/active_record/relation/batches.rb:51:in `each'
2018-04-03T17:41:52.700218096Z 	from /usr/local/bundle/gems/activerecord-4.2.10/lib/active_record/relation/batches.rb:51:in `block in find_each'
2018-04-03T17:41:52.700267681Z 	from /usr/local/bundle/gems/activerecord-4.2.10/lib/active_record/relation/batches.rb:124:in `find_in_batches'
2018-04-03T17:41:52.700290608Z 	from /usr/local/bundle/gems/activerecord-4.2.10/lib/active_record/relation/batches.rb:50:in `find_each'
2018-04-03T17:41:52.700312544Z 	from /app/lib/portus/background/security_scanning.rb:35:in `execute!'
2018-04-03T17:41:52.700334168Z 	from /app/bin/background.rb:57:in `block (2 levels) in <top (required)>'
2018-04-03T17:41:52.700356284Z 	from /app/bin/background.rb:55:in `each'
2018-04-03T17:41:52.700377727Z 	from /app/bin/background.rb:55:in `each_with_index'
2018-04-03T17:41:52.700399352Z 	from /app/bin/background.rb:55:in `block in <top (required)>'
2018-04-03T17:41:52.700422250Z 	from /app/bin/background.rb:54:in `loop'
2018-04-03T17:41:52.700443644Z 	from /app/bin/background.rb:54:in `<top (required)>'
2018-04-03T17:41:52.700465458Z 	from /usr/local/bundle/gems/railties-4.2.10/lib/rails/commands/runner.rb:60:in `load'
2018-04-03T17:41:52.700487493Z 	from /usr/local/bundle/gems/railties-4.2.10/lib/rails/commands/runner.rb:60:in `<top (required)>'
2018-04-03T17:41:52.700509670Z 	from /usr/local/bundle/gems/railties-4.2.10/lib/rails/commands/commands_tasks.rb:123:in `require'
2018-04-03T17:41:52.700531404Z 	from /usr/local/bundle/gems/railties-4.2.10/lib/rails/commands/commands_tasks.rb:123:in `require_command!'
2018-04-03T17:41:52.700687999Z 	from /usr/local/bundle/gems/railties-4.2.10/lib/rails/commands/commands_tasks.rb:90:in `runner'
2018-04-03T17:41:52.700729934Z 	from /usr/local/bundle/gems/railties-4.2.10/lib/rails/commands/commands_tasks.rb:39:in `run_command!'
2018-04-03T17:41:52.700752460Z 	from /usr/local/bundle/gems/railties-4.2.10/lib/rails/commands.rb:17:in `<top (required)>'
2018-04-03T17:41:52.700774927Z 	from bin/rails:12:in `require'
2018-04-03T17:41:52.700796561Z 	from bin/rails:12:in `<main>'
2018-04-03T17:42:10.296916711Z [Mailer config] Host:     registry.example.com
2018-04-03T17:42:10.297343796Z [Mailer config] Protocol: https://
2018-04-03T17:42:10.299411269Z [ruby] Using 2.5.1, but we recommend 2.5.0.
2018-04-03T17:42:27.750122519Z [Initialization] Running: 'Registry events', 'Security scanning', 'Registry synchronization'
2018-04-03T20:03:02.916818378Z /app/lib/portus/security.rb:49:in `vulnerabilities': undefined method `map' for nil:NilClass (NoMethodError)
2018-04-03T20:03:02.933769489Z 	from /app/lib/portus/background/security_scanning.rb:47:in `block in execute!'
2018-04-03T20:03:02.933829892Z 	from /usr/local/bundle/gems/activerecord-4.2.10/lib/active_record/relation/batches.rb:51:in `block (2 levels) in find_each'
2018-04-03T20:03:02.933872569Z 	from /usr/local/bundle/gems/activerecord-4.2.10/lib/active_record/relation/batches.rb:51:in `each'
2018-04-03T20:03:02.933896168Z 	from /usr/local/bundle/gems/activerecord-4.2.10/lib/active_record/relation/batches.rb:51:in `block in find_each'
2018-04-03T20:03:02.933949543Z 	from /usr/local/bundle/gems/activerecord-4.2.10/lib/active_record/relation/batches.rb:124:in `find_in_batches'
2018-04-03T20:03:02.933972701Z 	from /usr/local/bundle/gems/activerecord-4.2.10/lib/active_record/relation/batches.rb:50:in `find_each'
2018-04-03T20:03:02.934016622Z 	from /app/lib/portus/background/security_scanning.rb:35:in `execute!'
2018-04-03T20:03:02.934039569Z 	from /app/bin/background.rb:57:in `block (2 levels) in <top (required)>'
2018-04-03T20:03:02.934062297Z 	from /app/bin/background.rb:55:in `each'
2018-04-03T20:03:02.934084233Z 	from /app/bin/background.rb:55:in `each_with_index'
2018-04-03T20:03:02.934105907Z 	from /app/bin/background.rb:55:in `block in <top (required)>'
2018-04-03T20:03:02.934127873Z 	from /app/bin/background.rb:54:in `loop'
2018-04-03T20:03:02.934149356Z 	from /app/bin/background.rb:54:in `<top (required)>'
2018-04-03T20:03:02.934171332Z 	from /usr/local/bundle/gems/railties-4.2.10/lib/rails/commands/runner.rb:60:in `load'
2018-04-03T20:03:02.934193277Z 	from /usr/local/bundle/gems/railties-4.2.10/lib/rails/commands/runner.rb:60:in `<top (required)>'
2018-04-03T20:03:02.934215583Z 	from /usr/local/bundle/gems/railties-4.2.10/lib/rails/commands/commands_tasks.rb:123:in `require'
2018-04-03T20:03:02.934237949Z 	from /usr/local/bundle/gems/railties-4.2.10/lib/rails/commands/commands_tasks.rb:123:in `require_command!'
2018-04-03T20:03:02.934260376Z 	from /usr/local/bundle/gems/railties-4.2.10/lib/rails/commands/commands_tasks.rb:90:in `runner'
2018-04-03T20:03:02.934282010Z 	from /usr/local/bundle/gems/railties-4.2.10/lib/rails/commands/commands_tasks.rb:39:in `run_command!'
2018-04-03T20:03:02.934303996Z 	from /usr/local/bundle/gems/railties-4.2.10/lib/rails/commands.rb:17:in `<top (required)>'
2018-04-03T20:03:02.934326182Z 	from bin/rails:12:in `require'
2018-04-03T20:03:02.934347486Z 	from bin/rails:12:in `<main>'
2018-04-03T20:03:21.311121145Z [Mailer config] Host:     registry.example.com
2018-04-03T20:03:21.311237639Z [Mailer config] Protocol: https://
2018-04-03T20:03:21.313185388Z [ruby] Using 2.5.1, but we recommend 2.5.0.
2018-04-03T20:03:39.520358215Z [Initialization] Running: 'Registry events', 'Security scanning', 'Registry synchronization'
```

**Comment by mssola** [comment]
I believe this has already been fixed in #1778 as a side-effect. That being said, I've also traced some paths which look quite weak when it comes at fetching manifests. I'll take a look a look at this once I'm done with that PR.

Thanks for reporting :+1:

**Comment by Vad1mo** [comment]
true, the situation improved now.
